src/components/CreateProfile/LocationPreferences/LocationPreferences.tsx

```diff
@@ -41,37 +19,52 @@ const LocationPreferences = () => {
       </div>
       <div className={styles.right}>
         <div>
-          <TextInput
-            label="Country of residency"
-            placeholder="Start typing location"
-            value={country}
-            onChange={handleCountryChange}
-          />
+          <div className={errors.country ? styles.errorMsg : ''}>
+            <TextInput
+              label="Country of residency"
+              placeholder="Start typing location"
+              value={values.country}
+              onChange={handleChange}
+              name="country"
+            />
+            <p>{errors.country as string}</p>
+          </div>
+
           <CheckboxInput
             label="I’m open to residency relocation"
-            checked={openToRelocationCountry}
-            onChange={handleCheckboxChangeCountry}
+            checked={values.openToRelocationCountry}
+            onChange={handleChange}
+            name="openToRelocationCountry"
           />
         </div>
         <div>
-          <TextInput
-            label="City residency"
-            placeholder="Start typing location"
-            value={city}
-            onChange={handleCityChange}
-            addImportantIcon={true}
-          />
+          <div className={errors.city ? styles.errorMsg : ''}>
+            <TextInput
+              label="City residency"
+              placeholder="Start typing location"
+              value={values.city}
+              onChange={handleChange}
+              addImportantIcon={true}
+              name="city"
+            />
+            <p>{errors.city as string}</p>
```

Comment:
**nitpick:** We should not do type casting. What you will probably need to do is:

```jsx
{errors?.city && touched?.city &&  <p>{errors.city}</p>}
```